draft-ietf-quic-recovery.md

```diff
@@ -194,8 +194,14 @@ round-trip time (RTT) is critical to these algorithms and is described first.
 
 ## Computing the RTT estimate
 
-(To be filled)
+RTT is calculated when an ACK frame arrives by computing the difference
+between the current time and the time the largest newly acked packet was sent.
+If no packets are newly acknowledged, RTT cannot be calculated. When RTT is
+calculated, the ack delay field from the ACK frame SHOULD be subtracted from
+the RTT as long as the result is positive.
```

Comment:
Is this the right thing to do?
Sending an ack delay that would make the RTT negative sounds like a protocol violation. If we want to be generous with a peer that might just have adjusted its system time, the right thing to do would be ignoring that RTT sample completely, wouldn't it?

Reply:
It's a good question.  GQUIC does what I wrote, but I'm not sure it's the best thing.  

The only reason I wouldn't want to throw it out entirely is that it may cause you to get no RTT samples in the worst case.

Reply:
Your text is consistent with the pseudocode. Do we need a new issue to track this?

Reply:
Possibly.  Alternately, I can add a sentence indicating that samples may be ignored if the ack delay is too large?  I'm not convinced we will or need to come to consensus on this, so it might be better to allow either until we have a strong reason to suggest one over the other.

Reply:
Let's add this text in and open an issue. GQUIC currently subtracts this from RTT measurements, but if the peer's clock is off, this ack delay could be arbitrarily large, leading to a poor RTT estimate. Perhaps this should be protected by the measured minRTT as the min... but let's discuss that on an issue separately.